### DISCUSSION TITLE
Recovering Teslamate data

### DISCUSSION THREAD

**Discussion by alezanott** [teslamate-org/teslamate#2404] [Q&A]
I had some big issues with my a rpi 3b+ running Teslamate, so I decided to install everything from scratch on a new rpi 4.
I have recovered the Docker folder from the old SD card under /var/lib/docker. Now I want to use that old data in the newer Teslamate running on the rpi 4, as I want to keep track of all my previous charges. How can I substitute the right data and folders from the old docker folder to the new one?
Thanks in advance.

**Comment by ktsenvolt** [comment]
https://docs.teslamate.org/docs/maintenance/backup_restore
If have backup data from rpi 3b+ (teslamate.bck) you may transfer to rpi 4?

**Reply by alezanott** [reply]
> https://docs.teslamate.org/docs/maintenance/backup_restore If have backup data from rpi 3b+ (teslamate.bck) you may transfer to rpi 4?

Unfortunately I didn't have done a backup and it this step I have only the /var/lib/docker files copied from the SD of the rpi 3b+. 
Any idea of how I can use that folder on the new rpi 4?

**Reply by cwanja** [reply]
Have you tried just placing the contents from `/var/lib/docker` into your new Raspberry Pi 4? Tried research this a bit over the last couple of days and yet to find a good article on _how_ to do this. But find conflicting information that you can backup the volume and restore it versus you should not do that. Could try and **copy** it into that directory and if it goes south, just remove the contents.

**Reply by alezanott** [reply]
> Have you tried just placing the contents from `/var/lib/docker` into your new Raspberry Pi 4? Tried research this a bit over the last couple of days and yet to find a good article on _how_ to do this. But find conflicting information that you can backup the volume and restore it versus you should not do that. Could try and **copy** it into that directory and if it goes south, just remove the contents. 

Have tried creating a new container from scratch, then substituting all the files in that directory. Unfortunately I think that the files of Teslamate got corrupted doing this. When I tried connecting to Grafana I saw that all the stats had an error related to the missing files. I have tried this operation stopping docker-compose and docker, then running it again.

Maybe I don't have to create first the container of Teslamate? But actually if I don't create it I don't have the docker directory where I can substitute the files.

**Reply by alezanott** [reply]
> Have you tried just placing the contents from `/var/lib/docker` into your new Raspberry Pi 4? Tried research this a bit over the last couple of days and yet to find a good article on _how_ to do this. But find conflicting information that you can backup the volume and restore it versus you should not do that. Could try and **copy** it into that directory and if it goes south, just remove the contents.

Thank you for the reply!
I'm also still trying to find something that can help. Did you found anything interesting?

**Reply by cwanja** [reply]
I do not have anything else specifically in this realm. I do not have a dev box to try and port things over either. Sorry about that.

**Reply by alezanott** [reply]
> I do not have anything else specifically in this realm. I do not have a dev box to try and port things over either. Sorry about that.

No worries @cwanja !

**Comment by EasternPA** [comment]
I am also working on this. I have some pg_dumps from last July of last year and some raw file backups (like yours) from September. I lost everything and had to start over in October. I have a one month gap between the raw file backup and a 3 month gap to the last pg_dump before starting over. I tried to just merge my pg_dump from last July with a current pg_dump but the database server rejects all lines from the second import because the id numbers of the records in each table started over from 1, causing a conflict.

So now I'm stuck trying to manually merge all of the (important) tables by hand making sure the second instance of each table uses starts at a new id number that comes after the final id number used in the first instance. What a mess because I have millions of rows in my old table and hundreds of thousands in the new one. I can't use sed and awk to increment the id number in each row, so I'm still noodling on a solution.

If I can figure out how to mount the raw files like the ones you have and then do a pg_dump export, I'll come back and let you know what I've found. It would be nice to recover the two months' of lost-but-not-lost data. But even if I do, I'm still going to have to figure out how to merge two different backups where the tables start at row 1.

Update: I reconstructed all of the rows to combine both copies of each section into one and reserialized the ids of each row, but it still didn't work. I'm going to continue to live without the first year of data, unfortunately, and just get better about making and storing backups as time goes on.

**Reply by alezanott** [reply]
> If I can figure out how to mount the raw files like the ones you have and then do a pg_dump export, I'll come back and let you know what I've found

That would be super awesome!

> I'm going to continue to live without the first year of data, unfortunately, and just get better about making and storing backups as time goes on

Basically a pretty close situation as mine, I had lost the first year of data of my Tesla. 
I wanted program automatic backups on a NAS on the same network of the RPI. Do you have any clue in doing this?

Thank you in advance @EasternPA

**Reply by EasternPA** [reply]
> I wanted program automatic backups on a NAS on the same network of the RPI. Do you have any clue in doing this?

My plan was to take a slightly different approach. I installed OpenMediaVault on a Pi, installed the docker module, and I run TeslaMate in that docker-compose. I eventually plan to establish a share in OMV that would be mounted by TM. That would make OMV the native storage location for the TM files. If I can't successfully pass through the storage, I may share it via SMB or NFS all within the same host. It's not like we're sending terabytes of data over the course of the day. The data is stored outside OMV for now, so I'm still in the early stages.

Once you get TM to natively store data within the OMV share (or NFS export), you're getting into native NAS-like tools and features to keep it backed up. I also need to learn more about Kubernetes, DevOps and GitOps for work, so this stack presents a great opportunity to learn those technologies and slowly build your (my) skills in this space over time.

Update: I forgot to mention periodically exporting the data using the commands from the backup/restore page in the documentation. Accessing the raw files from the filesystem like the ones you have has just not worked out.

**Comment by ktsenvolt** [comment]
Automated backup to Google Drive #2385


Ken Tsen  

    On Tuesday, March 8, 2022, 08:54:14 AM EST, alezanott ***@***.***> wrote:  
 
 



If I can figure out how to mount the raw files like the ones you have and then do a pg_dump export, I'll come back and let you know what I've found


That would be super awesome!


I'm going to continue to live without the first year of data, unfortunately, and just get better about making and storing backups as time goes on


Basically a pretty close situation as mine, I had lost the first year of data of my Tesla.
I wanted program automatic backups on a NAS on the same network of the RPI. Do you have any clue in doing this?

Thank you in advance @EasternPA

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ktsenvolt** [comment]
Automated backup to Google Drive #2385
